qiime2/sdk/typemap.py

```diff
@@ -0,0 +1,48 @@
+import random
+from itertools import combinations
+import networkx as nx
+import matplotlib.pyplot as plt
+
+import qiime 2
+
+q2methods = qiime2.sdk.util.actions_by_input_type('FeatureTable[Frequency]')
+
+#############testing##################
+q2p = qiime2.sdk.PluginManager()
+
+#for plugin, value in q2p.plugins.items():
+#    print(value.methods.keys())
+##############testing#################
+
+
+
+class TypeMap:
```

Comment:
We probably want to think of a new name for this ---`TypeMap` is already taken in [this PR](https://github.com/qiime2/qiime2/pull/444) that @ebolyen is working on. Historically we have used this term in QIIME 2 to refer to a mapping of outputs that belong to a specific set of inputs and parameters.

I don't actually see this code being tested anywhere in your tests above --- can you comment on this?